src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,48 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_LIST_SIZE = 10;
+    private static final int START_SIZE_VALUE = 1;
+    private StoragePair<K,V>[] list;
+    private int size = 0;
+
+    public StorageImpl() {
+        list = new StoragePair[MAX_LIST_SIZE];
+    }
+
     @Override
     public void put(K key, V value) {
+        size = START_SIZE_VALUE;
+        StoragePair<K,V> tempStorage = new StoragePair<>(key, value);
+        for (int i = 0; i < list.length; i++) {
+            if (list[i] != null) {
+                size++;
+            }
+            if (list[i] == null || valid(i, key)) {
+                list[i] = tempStorage;
+                break;
+            }
+        }
+
     }
 
     @Override
     public V get(K key) {
+        for (int i = 0; i < list.length; i++) {
+            if (valid(i, key)) {
+                return list[i].getValue();
+            }
+        }
         return null;
     }
 
     @Override
     public int size() {
-        return -1;
+        return size;
+    }
+
+    private boolean valid(int i, K key) {
```

Comment:
let's make this method receive a storagePair and key.
```suggestion
    private boolean isKeyInPair(StoragePair pair, K key) {
```